vaadin-spreadsheet/src/main/java/com/vaadin/addon/spreadsheet/Spreadsheet.java

```diff
@@ -1010,40 +1010,40 @@ public void attach() {
      * If the currently active sheet is set hidden, another sheet is set as
      * active sheet automatically. At least one sheet should be always visible.
      * 
-     * @param hidden
-     *            Visibility state to set: 0-visible, 1-hidden, 2-very hidden.
      * @param sheetPOIIndex
      *            Index of the target sheet within the POI model, 0-based
+     * @param visibility
+     *            Visibility state to set: visible, hidden, very hidden.
      * @throws IllegalArgumentException
      *             If the index or state is invalid, or if trying to hide the
      *             only visible sheet.
      */
-    public void setSheetHidden(int sheetPOIIndex, int hidden)
+    public void setSheetHidden(int sheetPOIIndex, SheetVisibility visibility) 
             throws IllegalArgumentException {
         // POI allows user to hide all sheets ...
-        if (hidden != 0
+        if (visibility != SheetVisibility.VISIBLE
                 && SpreadsheetUtil.getNumberOfVisibleSheets(workbook) == 1
-                && !workbook.isSheetHidden(sheetPOIIndex)) {
+                && ! (workbook.isSheetHidden(sheetPOIIndex)
+                        || workbook.isSheetVeryHidden(sheetPOIIndex))) {
             throw new IllegalArgumentException(
                     "At least one sheet should be always visible.");
         }
         boolean isHidden = workbook.isSheetHidden(sheetPOIIndex);
         boolean isVeryHidden = workbook.isSheetVeryHidden(sheetPOIIndex);
         int activeSheetIndex = workbook.getActiveSheetIndex();
-        SheetVisibility visibility = isVeryHidden ? SheetVisibility.VERY_HIDDEN
-                : (isHidden ? SheetVisibility.HIDDEN : SheetVisibility.VISIBLE);
```

Comment:
This assignment completely ignores the input value, and just sets the sheet state to its current state, a no-op.  No unit tests reference this, and apparently no one is using it, as it is an obvious bug once you try to hide sheets.

My main purpose was to change the API to use POI Enums instead of ints, but then this jumped out at me too, so I fixed it.